### ISSUE TITLE
Dynamic popup links cause Undefined index issues with Elementor 2.8.1 / Pro 2.7.3

### ISSUE THREAD

**Issue by oddlots** [elementor/elementor#9847]
## Prerequisites
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

After upgrading to Elementor 2.8.1, wp_debug repeatedly reports an issue with an undefined index in tags.php when using a dynamic link to a popup.

> Notice: Undefined index: before in .../public_html/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 41
> 
> Notice: Undefined index: after in .../public_html/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 45

## Steps to reproduce

1. Have a site with Elementor 2.8.1 and Elementor Pro 2.7.3 installed and activated and be running with the wp_debug flag set to true. (theme is irrelevant) 
2. Add a new Empty Page and edit with Elementor
3. Drag an Icon onto the page and use the "dynamic" dropdown to select "popup"
4. Preview changes and you'll see the "Undefined index: fallback" warnings.

notes: what you use for the link is irrelevant. wether you have an popup to open is irrelevant.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MySQL Community Server (GPL) v5.6.46
	PHP Version: 7.1.32
	PHP Max Input Vars: 1000
	PHP Max Post Size: 32M
	GD Installed: Yes
	ZIP Installed: No
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.3
	Site URL: XXXXXXXXXXXX
	Home URL: XXXXXXXXXXXX
	WP Multisite: No
	Max Upload Size: 32 MB
	Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en
	Timezone: 0
	Debug Mode: Active

== Theme ==
	Name: Twenty Twenty
	Version: 1.0
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Safari/605.1.15

== Active Plugins ==
	Elementor
		Version: 2.8.1
		Author: Elementor.com

	Elementor Pro
		Version: 2.7.3
		Author: Elementor.com


== Elements Usage ==
	
	header : 2
		icon-list : 1
		image : 1
		nav-menu : 2
		search-form : 1
	section : 1
		divider : 1
		heading : 2
		image : 1
		nav-menu : 1
		theme-site-logo : 1
	single : 2
		bordered-image : 1
		bordered-text-editor : 1
	wp-post : 0
		button : -2
		divider : 1
		form : 2
		heading : 24
		icon-list : 1
		image : 3
		image-carousel : 1
		nav-menu : 1
		social-icons : 1
		spacer : 1
		text-editor : 5
		theme-page-title : 1
		theme-site-logo : 1
		video : 1
	wp-page : 10
		icon : 1


== Log ==
	
Log: showing 20 of 392019-06-23 21:51:06 [info] Elementor Pro/Upgrades - _v_2_5_4_form Finished 
2019-06-23 21:51:06 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.4.4',
  'to' => '2.5.9',
)]
2019-08-09 01:02:18 [info] elementor::elementor_updater Started 
2019-08-09 01:02:18 [info] Elementor/Upgrades - _v_2_6_0_fa4_migration_flag Start  
2019-08-09 01:02:18 [info] Elementor/Upgrades - _v_2_6_0_fa4_migration_flag Finished 
2019-08-09 01:02:18 [info] Elementor/Upgrades - _v_2_6_6_fa4_migration_button Start  
2019-08-09 01:02:19 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.16',
  'to' => '2.6.8',
)]
2019-08-09 01:02:21 [info] Elementor/Upgrades - _v_2_6_6_fa4_migration_button Finished 
2019-08-09 01:02:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.16',
  'to' => '2.6.8',
)]
2019-08-09 01:02:43 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.9',
  'to' => '2.6.2',
)]
2019-12-03 18:40:29 [info] elementor::elementor_updater Started 
2019-12-03 18:40:29 [info] Elementor/Upgrades - _v_2_7_1_remove_old_usage_data Start  
2019-12-03 18:40:29 [info] Elementor/Upgrades - _v_2_7_1_remove_old_usage_data Finished 
2019-12-03 18:40:29 [info] Elementor/Upgrades - _v_2_7_1_recalc_usage_data Start  
2019-12-03 18:40:30 [info] Elementor/Upgrades - _v_2_7_1_recalc_usage_data Finished 
2019-12-03 18:40:30 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.6.8',
  'to' => '2.7.5',
)]
2019-12-03 18:40:30 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '2.6.8',
  'to' => '2.7.5',
)]
2019-12-03 18:40:39 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.6.2',
  'to' => '2.7.3',
)]
2019-12-10 00:10:44 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.7.5',
  'to' => '2.7.6',
)]
2019-12-11 16:51:45 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.7.6',
  'to' => '2.8.1',
)]

PHP: showing 1 of 1

PHP: 2019-12-11 17:28:33 [notice X 13][/home/lilolbear/public_html/c/spaamfaa/wp-content/plugins/elementor/core/dynamic-tags/tag.php::53] Undefined index: fallback [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
```
</details>

**Comment by shilo-ey** [comment]
Thanks for reporting and contributing.

We are examining this issue and will release a patch ASAP.

**Comment by beniciot** [comment]
Any update on this bug? I'm having the same issue;/

**Comment by crospyder** [comment]
Same here - anyone - ideas? - every site using Elementor with this vrsion and using button with popup just got stuck getting this same error - we need solution asap - thnx

**Comment by swinggraphics** [comment]
Throw some `isset( $settings[…] && …` into the conditional statements on the affected lines for now to solve the warnings until an update is released.

**Comment by beniciot** [comment]
> Throw some `isset( $settings[…] && …` into the conditional statements on the affected lines for now to solve the warnings until an update is released.

Thanks! Do you mean that I should add it to all the lines affected below ? 

`
Notice: Undefined index: after in /var/www/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 45

Notice: Undefined index: before in /var/www/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 41

Notice: Undefined index: after in /var/www/wp-content/pluginsNotice: Undefined index: after in /var/www/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 45

Notice: Undefined index: before in /var/www/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 41

Notice: Undefined index: after in /var/www/wp-content/plugins/elementor/core/dynamic-tags/tag.php on line 45/elementor/core/dynamic-tags/tag.php on line 45
`

**Comment by crospyder** [comment]
we did the same... all day turnibg off popups all arround sites.. still not done tho 🤕

Neven Pausić

________________________________

**Comment by swinggraphics** [comment]
@beniciot Make these changes to tag.php:
line 41
`if ( isset( $settings['before'] ) && ! Utils::is_empty( $settings['before'] ) ) {`

line 45
`if ( isset( $settings['after'] ) && ! Utils::is_empty( $settings['after'] ) ) {`

line 53
`} elseif ( isset( $settings['fallback'] ) && ! Utils::is_empty( $settings['fallback'] ) ) {`

**Comment by manoj-identixweb** [comment]
@swinggraphics 
Thanks for corrections in that issue. Looking forward to the new update in elementor in which this issue will be fixed.

**Comment by lanimal75** [comment]
my site https://www.dechautstetten.com/ runs on wp 5.3.1 and elementor pro 2.7.3 using PHP 7.3 and I had the same problem. I downgraded into PHP 7.2 and the problem is solved. I guess i will have to wait untill a new version of elementor pro comes out...

**Comment by colbyalbo** [comment]
I can confirm that this has been fixed on my installs of v2.8.2

**Comment by swinggraphics** [comment]
Yes, I see they rejiggered the `Utils::is_empty` method to accommodate.

**Comment by beniciot** [comment]
> I can confirm that this has been fixed on my installs of Elementor v2.8.2

That's great news! Thank you

**Comment by bamcguire** [comment]
I am on the cutlrrent version but I am unable to edit pop-ups.  I can insert templates but I cannot interact with any of the elements.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v2.8.2

Feel free to update

Thanks!
